dbms/src/Common/FiberPool.h

```diff
@@ -0,0 +1,339 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Source: https://github.com/moneroexamples/fiberpool
+// FiberPool based on:
+// http://roar11.com/2016/01/a-platform-independent-thread-pool-using-c14/
+
+#pragma once
+
+#include <Common/BitHelpers.h>
+#include <Common/ExecutableTask.h>
+#include <Common/setThreadName.h>
+
+#include <boost/fiber/all.hpp>
+
+inline thread_local bool g_run_in_fiber = false;
+
+inline void adaptiveYield()
+{
+#ifdef TIFLASH_USE_FIBER
+    if (g_run_in_fiber)
+        boost::this_fiber::yield();
+    else
+        std::this_thread::yield();
```

Comment:
Why does it need to `yield` a system thread here?

Reply:
just for consistency, it has little cost even for an OS thread.